lib/api/apiUtils/object/partInfo.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Checks for partNumber in request query and returns the part number
+ * @param {object} query - request query
+ * @return {(Integer|undefined)} - part number, zero if part number is less than
+ * 1 and undefined if no partNumber is found in the request query
+ */
+function getPartNumber(query) {
+    if (query && query.partNumber !== undefined) {
+        return Number.isNaN(query.partNumber) ?
+            0 : Number.parseInt(query.partNumber, 10);
+    }
+    return undefined;
+}
+
+/**
+ * Gets the size of the requested part of the object
+ * @param {object} objMD - object metadata
+ * @param {object} partNumber - part number
+ * @return {(Integer|undefined)} - size of the part or undefined
+ */
+function getPartSize(objMD, partNumber) {
+    let size;
+    if (partNumber && objMD && objMD.location
+        && objMD.location.length >= partNumber) {
+        const locations = [];
+        for (let i = 0; i < objMD.location.length; i++) {
```

Comment:
I have a small performance concern here, I'm not sure it's worth trying to optimize this depending on the use case for this function and the associated performance hit, but want to raise the point anyway for brainstorming. This is something that might be worth testing with a specific unit test for example, or even on a lab, to assess the impact.

If the MPU is large, i.e. close to 10000 parts (which can sometimes give more than 10000 locations as we observed in the past, but should not generally go beyond 20K-30K I think), and if the request is for a part number towards the end of the locations array, we might have to loop several thousand times before reaching the part we want. Considering this blocks the JS process entirely during that time, it could be significant enough to slow down everything a little bit (depending on how many such HEAD requests arrive). Also, if one wants to do a HEAD to know the size of each single part in a large MPU for example, it might incur significant load because of this.

One way to improve might be possible by using dichotomy to search since I think the locations array will always be sorted by part number, although multiple locations can refer to the same part number. But we would have to double-check first the assumption that the array is sorted by part number.